wgpu/backends/wgpu_native/_api.py

```diff
@@ -2276,15 +2295,35 @@ def draw_indexed(
 
     def draw_indexed_indirect(self, indirect_buffer, indirect_offset):
         buffer_id = indirect_buffer._internal
-        # H: void f(WGPURenderPassEncoder renderPassEncoder, WGPUBuffer indirectBuffer, uint64_t indirectOffset)
-        libf.wgpuRenderPassEncoderDrawIndexedIndirect(
-            self._internal, buffer_id, int(indirect_offset)
-        )
+        function = self.__pick_render_function(
+            # H: void f(WGPURenderPassEncoder renderPassEncoder, WGPUBuffer indirectBuffer, uint64_t indirectOffset)
+            render_pass=libf.wgpuRenderPassEncoderDrawIndexedIndirect,
+            # H: void f(WGPURenderBundleEncoder renderBundleEncoder, WGPUBuffer indirectBuffer, uint64_t indirectOffset)
+            render_bundle=libf.wgpuRenderBundleEncoderDrawIndexedIndirect,
+        )
+        function(self._internal, buffer_id, int(indirect_offset))
```

Comment:
You're now using two different approaches to select the correct function. Can we use the set-function-as-class-attribute approach here as well? I like that one better, TBH

Reply:
Sorry, I jumped right into the code and read your post later.

Good point about the annotations. I think its well worth having these near the code that does the call. So please go with the `__pick_render_function` approach 👍

Reply:
If I'm reading between the lines correctly, you think the `type(self)._xxx_yyy_function` implementation is cleaner, but you don't like that the annotations are missing when we do that.  I think I've come up with a good hybrid approach that gives me both.  I'll be updating this pull in a few hours.  

If after seeing the hybrid approach, you still prefer `self.__pick_render_function`, etc, then I'll change it back.

Reply:
If I'm reading between the lines correctly, you think the `type(self)._xxx_yyy_function` implementation is cleaner, but you don't like that the annotations are missing when we do that.  I think I've come up with a good hybrid approach that gives me both.  I'll be updating this pull in a few hours.  

If after seeing the hybrid approach, you still prefer `self.__pick_render_function`, etc, then I'll change it back.